The script does not expect use of ccache, hence fix it to recognise it.

2010/08/17
Nitin A Kamble <nitin.a.kamble@intel.com>

Upstream-Status: Inappropriate [configuration]

Index: valgrind-3.5.0/coregrind/link_tool_exe_linux.in
===================================================================
--- valgrind-3.5.0.orig/coregrind/link_tool_exe_linux.in
+++ valgrind-3.5.0/coregrind/link_tool_exe_linux.in
@@ -62,6 +62,11 @@ die "Bogus alt-load address"
 
 # The cc invokation to do the final link
 my $cc = $ARGV[1];
+my $next = 2;
+if ("$cc" eq "ccache") {
+    $cc = "$cc $ARGV[2]";
+    $next = 3;
+}
 
 # and the 'restargs' are argv[2 ..]
 
@@ -71,7 +76,7 @@ my $cc = $ARGV[1];
 my $cmd="$cc -static -Wl,-Ttext=$ala";
 
 # Add the rest of the parameters
-foreach my $n (2 .. $#ARGV) {
+foreach my $n ($next .. $#ARGV) {
    $cmd = "$cmd $ARGV[$n]";
 }
 
